### PR TITLE
Rabbitmq Helm Chart

### PR DESCRIPTION
**Pull request by karun-singh**
- Uses the bitnami rabbitmq helm chart

### CODE REVIEW

**Comment by karun-singh** [comment]
- [ ] Add an autoscaling script to replace the HPA.

File: K8s-deployment/Charts/databroker/values.yaml [hunk 1]
```diff
@@ -0,0 +1,1302 @@
...
+  ## @param clustering.enabled Enable RabbitMQ clustering
+  ##
+  enabled: true
+  ## @param clustering.addressType Switch clustering mode. Either `ip` or `hostname`
+  ##
+  addressType: hostname
+  ## @param clustering.rebalance Rebalance master for queues in cluster when new replica is created
+  ## ref: https://www.rabbitmq.com/rabbitmq-queues.8.html#rebalance
+  ##
+  rebalance: false
```
Comment (abhi4578) [inline K8s-deployment/Charts/databroker/values.yaml:223]: make this true. want the queues master to be rebalanced on addition of replicas

File: K8s-deployment/Charts/databroker/values.yaml [hunk 2]
```diff
@@ -0,0 +1,1302 @@
...
+  management.ssl.keyfile = /opt/bitnami/rabbitmq/certs/server_key.pem
+  disk_free_limit.relative = 1.0
+  vm_memory_high_watermark_paging_ratio = 0.6
+  # Logging
+  log.console = true
+  log.console.level = info
+  log.file = false
+  # Monitoring
+  collect_statistics_interval = 10000
+  #prometheus.tcp.port = 15692
```
Comment (abhi4578) [inline K8s-deployment/Charts/databroker/values.yaml:407]: why is this commented?

File: K8s-deployment/Charts/databroker/example-aws-resource-values.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  node.kubernetes.io/instance-type: c5a.large
+resources:
+  limits:
+    cpu: "1800m"
+    memory: "3.8Gi"
+  requests:
+    cpu: "1500m"
+    memory: "3Gi"
+persistence:
+  storageClass: "ebs-storage-class"
```
Comment (abhi4578) [inline K8s-deployment/Charts/databroker/example-aws-resource-values.yaml:11]: ebs-csi-storage-class

File: K8s-deployment/Charts/databroker/external-client-service.yaml [hunk 4]
```diff
@@ -3,14 +3,14 @@ apiVersion: v1
 metadata:
   annotations:
     # For aws service
-    service.beta.kubernetes.io/aws-load-balancer-type: "nlb"
+    #service.beta.kubernetes.io/aws-load-balancer-type: "nlb"
```
Comment (abhi4578) [inline K8s-deployment/Charts/databroker/external-client-service.yaml:6]: Can have two files 
1) external-client-aws-service.yaml
2) external-client-azure-service.yaml

File: K8s-deployment/Charts/databroker/README.md [hunk 5]
```diff
@@ -0,0 +1,442 @@
...
+```console
+ cp -r example-secrets/secrets .
+```
+
+```
+# secrets directory after generation of secret files
+```sh
+secrets/
+├── credentials
+|   ├── .erlang.cookie (Random characters - 50)
```
Comment (abhi4578) [inline K8s-deployment/Charts/databroker/README.md:16]: rabbitmq-erlang-cookie or .erlang.cookie

File: K8s-deployment/Charts/databroker/README.md [hunk 6]
```diff
@@ -0,0 +1,442 @@
...
+helm uninstall rabbitmq -n rabbitmq
+```
+The command removes all the Kubernetes components associated with the chart and deletes the release.
+
+
+To delete rabbitmq 'pvc'
+```console
+kubectl delete pvc data-rabbitmq-X -n rabbitmq
+```
+## To dos:
```
Comment (abhi4578) [inline K8s-deployment/Charts/databroker/README.md:82]: These can be removed

File: K8s-deployment/Charts/databroker/README.md [hunk 7]
```diff
@@ -0,0 +1,442 @@
...
+2. Prometheus operator integration with RMQ monitoring
+3. Differentiate following things between environements - dev(mostly deployed in minikube), test and production mostly using Kustomize:
+   - Persistent Volumes (PV)  
+   - resource requests and limits
+   - namespace naming
+## References
+1. [RabbitMQ in k8s using StatefulSet in Minikube](https://github.com/rabbitmq/diy-kubernetes-examples/tree/master/minikube)
+2. [Rabbitmq in k8s using Statefulset in cloud-specific](https://github.com/rabbitmq/diy-kubernetes-examples/tree/master/gke)
+3. [Cluster Autoscaler](https://github.com/kubernetes/autoscaler/tree/master/cluster-autoscaler/cloudprovider/aws)
+
```
Comment (abhi4578) [inline K8s-deployment/Charts/databroker/README.md:93]: You can just point to bitnami helm chart . No need to have params here

**Review by abhi4578 — CHANGES_REQUESTED** [review]

File: K8s-deployment/Charts/databroker/README.md [hunk 8]
```diff
@@ -0,0 +1,442 @@
...
+## Parameters
```
Comment (abhi4578) [inline K8s-deployment/Charts/databroker/README.md:94]: You can just point to bitnami helm chart . No need to have params here

File: K8s-deployment/Charts/databroker/values.yaml (hunk 1 above)
Comment (karun-singh) [reply K8s-deployment/Charts/databroker/values.yaml]: Done

File: K8s-deployment/Charts/databroker/values.yaml (hunk 2 above)
Comment (karun-singh) [reply K8s-deployment/Charts/databroker/values.yaml]: It is already defined above in the metrics configuration. 
Removed commented line.

File: K8s-deployment/Charts/databroker/example-aws-resource-values.yaml (hunk 3 above)
Comment (karun-singh) [reply K8s-deployment/Charts/databroker/example-aws-resource-values.yaml]: Done

File: K8s-deployment/Charts/databroker/external-client-service.yaml (hunk 4 above)
Comment (karun-singh) [reply K8s-deployment/Charts/databroker/external-client-service.yaml]: Split into 2 files for aws and azure and added steps in README to make an `external-client-service.yaml` from these, which is referenced in the install script in the kubectl command.

File: K8s-deployment/Charts/databroker/README.md (hunk 5 above)
Comment (karun-singh) [reply K8s-deployment/Charts/databroker/README.md]: Fixed

File: K8s-deployment/Charts/databroker/README.md (hunk 6 above)
Comment (karun-singh) [reply K8s-deployment/Charts/databroker/README.md]: Removed

File: K8s-deployment/Charts/databroker/README.md (hunk 8 above)
Comment (karun-singh) [reply K8s-deployment/Charts/databroker/README.md]: Removed.

**Comment by abhi4578** [comment]
backport this PR to 4.0.0 branch @karun-singh

**Review by abhi4578 — APPROVED** [review]
LGTM!

**Pull request merged**